### PR TITLE
Add PHP 8.0 support

### LINKED ISSUE #9: PHP 8.0 support
### Feature Request

|    Q        |   A
|------------ | ------
| New Feature | yes

#### Summary

To be prepared for the december release of PHP 8.0, this repository has some additional TODOs to be tested against the new major version.

In order to make this repository compatible, one has to follow these steps:

- [ ] Modify `composer.json` to provide support for PHP 8.0 by adding the constraint `~8.0.0`
- [ ] Modify `composer.json` to drop support for PHP less than 7.3
- [ ] Modify `composer.json` to implement phpunit 9.3 which supports PHP 7.3+
- [ ] Modify `.travis.yml` to ignore platform requirements when installing composer dependencies (simply add `--ignore-platform-reqs` to `COMPOSER_ARGS` env variable)
- [ ] Modify `.travis.yml` to add PHP 8.0 to the matrix (*NOTE*: Do not allow failures as PHP 8.0 has a feature freeze since 2020-08-04!)
- [ ] Modify source code in case there are incompatibilities with PHP 8.0

### PR DESCRIPTION
**Pull request by bfoosness**
|    Q          |   A
|-------------- | ------
| New Feature   | yes

### Description

Close #9

### CODE REVIEW

File: test/View/Helper/FlashMessengerTest.php [hunk 1]
```diff
@@ -376,6 +381,7 @@ public function testCanDisplayListOfCurrentMessagesCustomisedByConfigSeparator()
         $helperPluginManager = $services->get('ViewHelperManager');
         $helper              = $helperPluginManager->get('flashmessenger');
 
+        $helper->setView(new PhpRenderer());
```
Comment (bfoosness) [inline test/View/Helper/FlashMessengerTest.php]: Resolves

```LaminasTest\Mvc\Plugin\FlashMessenger\View\Helper\FlashMessengerTest::testCanDisplayListOfCurrentMessagesCustomisedByConfigSeparator
TypeError: method_exists(): Argument #1 ($object_or_class) must be of type object|string, null given

/usr/src/myapp/vendor/laminas/laminas-view/src/Helper/FlashMessenger.php:344
/usr/src/myapp/vendor/laminas/laminas-view/src/Helper/FlashMessenger.php:165
/usr/src/myapp/vendor/laminas/laminas-view/src/Helper/FlashMessenger.php:128
/usr/src/myapp/test/View/Helper/FlashMessengerTest.php:384
```

Please advise if another solution is preferred.

**Comment by fezfez** [comment]
can it be merge ?

**Comment by bfoosness** [comment]
> can it be merge ?

laminas-json has just been updated (which is an indirect dependency of this module via laminas-view), so I believe this is now ready.

**Comment by froschdesign** [comment]
@bfoosness 
Could you rebase against the branch 1.3.x? This will use the new CI workflow.
Thanks in advance! 👍

**Comment by bfoosness** [comment]
I also updated all Composer dependencies to latest so the new CI check against the lock file would pass.

File: test/View/Helper/FlashMessengerTest.php (hunk 1 above)
Comment (froschdesign) [reply test/View/Helper/FlashMessengerTest.php]: @bfoosness
Please check #11

Comment (bfoosness) [reply test/View/Helper/FlashMessengerTest.php]: I updated to reflect a similar change to https://github.com/laminas/laminas-view/blob/3.0.x/src/Helper/FlashMessenger.php#L344

Currently tests are failing when we use the lowest dependencies for PHP 8 (https://github.com/laminas/laminas-mvc-plugin-flashmessenger/pull/10/checks?check_run_id=2313255852) due to the same line above at version 2.10 of laminas-view: https://github.com/laminas/laminas-view/blob/2.10.0/src/Helper/FlashMessenger.php#L344

Comment (bfoosness) [reply test/View/Helper/FlashMessengerTest.php]: Is the solution to bump the minimum version of laminas-view to 2.12, which is when it was fixed? https://github.com/laminas/laminas-view/blob/2.12.0/src/Helper/FlashMessenger.php#L344

Comment (bfoosness) [reply test/View/Helper/FlashMessengerTest.php]: Why _does_ this class live in both this repository and laminas-view?

Comment (Xerkus) [reply test/View/Helper/FlashMessengerTest.php]: BC reasons. It will be removed when next major is released

Comment (Xerkus) [reply test/View/Helper/FlashMessengerTest.php]: Yes, bump the laminas-view but also open issue about FlashMessager helper from laminas-view being preferred over the one provided by this package. This is either a test setup issue or a legit problem that needs to be investigated for a potential fix.

Comment (bfoosness) [reply test/View/Helper/FlashMessengerTest.php]: I thought as much. Unfortunately it looks like laminas-view 2.12 requires PHP 8, so by making that the minimum we'd be forcing PHP 8 to use this package?

Comment (bfoosness) [reply test/View/Helper/FlashMessengerTest.php]: Never mind; it allows ^7.3, too.

Comment (bfoosness) [reply test/View/Helper/FlashMessengerTest.php]: It looks like https://github.com/laminas/laminas-view/issues/16 already exists. Does that cover us here?

**Pull request merged**